### PR TITLE
Alerting: UI standardization and fixes

### LINKED ISSUE #34051: Alerting: UI fixes
_Tested on Dev env. Latest tested on May 13_

### Alert rules listing - - - - - - - - - - - - -   

**Functionality**
- Missing option to add a new group 
- Missing expand all button

**UI**

- :heavy_check_mark: Search input missing placeholder
- :heavy_check_mark: State filter is missing label
- :heavy_check_mark: View as should be a radio button group - same as state. It should also be on he same line
- Title of two areas (treshold and system) is missing the word “alerts” at the end
- :heavy_check_mark: Update table rows content to match Figma - remove status, add summary, add timing
    Domas: partially-  I kept status, but renamed to `health`. It seems that both health and state are important to see
- :heavy_check_mark: update sections from Treshold and System or application to the new names
-  NoData as a state should be displayed with an orange label not a blue one

### Alert rule creation  - - - - - - - - - - - - -   

**Functionality**

- I want to save a newly created alert (prometheus one), and I get an errro message saying “There are errors in the form below. Please fix them and try saving again” but there is no indicator of where those errors are  

**UI**
- :heavy_check_mark: Title of each section is too big, should be H4 - 16px
- :heavy_check_mark: update text for alert types to Grafana managed alerts and Cortex/Loki managed alerts - see Figma
- :heavy_check_mark: Namespace and group selects should be on the same line
- :heavy_check_mark: make sure all those inputs and selects from 1st step have equal width
- :heavy_check_mark: spacing between inputs/selects on the same row should be 24px
- :heavy_check_mark: add more vertical spacing between each section - that spacing should be 56px
- :heavy_check_mark: as agreed, in section 4, we should show from start summary, description and runbook URL
- missing section 5, where we tell the user about notifications policies 
- use lighter gray for steps count background (1…5)

### Silences - - - - - - - - - - - - -   

**Functionality**
- :heavy_check_mark:I get 2 messages every time I save a silence “Silence created” and. “Silence updated” - feels redundant 

**UI** 
- :heavy_check_mark: missing duration input field as we have today - it’s useful as it’s faster to read by a human, than a date range is
-  :heavy_check_mark: Update “matchers” to Matching labels - as matcher is not a real word
-  :heavy_check_mark: Update label of first input from matchers to say “Label” instead of “Name”
- All inputs should have placeholders

### Contact points - - - - - - - - - - - - -   

**Functionality**
- as soon as I saved my first contact point, in the table I see now one called default, that doesn’t have a type. this seems like a bug
  Domas: canot repro :/ 

**UI**
- create message template add spacing between title and form
- inputs should have placeholders
- indicate with * required fields (both for template form and contact point form - really important for slack for example, super long list of inputs)
- for contact points that have long list of optional fields we should show that under an expandable element, like we have today in Grafana OSS
- Contact points - icons on the buttons (duplicate and delete) should be 16x16px 

### Routes- - - - - - - - - - - - -   

**UI** 
- :heavy_check_mark: update tab name to Notification policies 
- Remove border bottom from root policy box
- Root policy edit - create a contact point link should have underline 
- timing options 
    - remove box surrounding this area -please see Figma. 
    - also caret icon should point to the right when timings are collapsed, and down when they are expanded
    - timing options should have defaults values
- Specific policies edit/create
    -   :heavy_check_mark: Please update text from “Matchers” to “Matching labels” as matchers is not a real word. Same for the button - see Figma, is updated
    - :heavy_check_mark: we should add a delete icon for each row (as we have in silences), my bad, I forgot to add it in Figma
    - Fix input validations - the tooltip appears inline instead of being positioned absolute. if only one input per row has a validation error everything gets shifted down (see screenshot)
    - On/Off switches should have the label on the same line as the switch 
Table listing of specific policies - don’t show nested policies table if there is no value in there. Most of these policies will be on one level only

### General issue - - - - - - - - - - - - - 

* Hard to silence an alert - no way to do it from alert listing, and no way to see affected alerts by a silence, so basically has to try it in blind, or to remember all the labels
* Try to standardize parts of the UI - indicate required fields (we indicate them only on some forms), Empty states for tables - they should all look the same (like the empty state from specific policies under notification policies)

Screenshot 2021-05-13 at 13 10 25
Screenshot 2021-05-13 at 15 22 09
Screenshot 2021-05-13 at 15 25 01

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

This PR adds some new functionality and helps standardize the UI on a few pieces
- New: adds an expand/collapse feature for rule groups
- New: adds a filter for rule type on the rules page
- Uses dynamic table on the Silences page
- Adds placeholder text for inputs in the silences form

**Which issue(s) this PR fixes**:

Fixes #34051, #35496, #35715

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/components/amroutes/AmRootRouteForm.tsx [hunk 1]
```diff
@@ -104,7 +108,7 @@ export const AmRootRouteForm: FC<AmRootRouteFormProps> = ({
                 <div className={cx(styles.container, styles.timingContainer)}>
                   <InputControl
                     render={({ field, fieldState: { invalid } }) => (
-                      <Input {...field} className={styles.smallInput} invalid={invalid} />
+                      <Input {...field} className={styles.smallInput} invalid={invalid} placeholder={'Time'} />
```
Comment (domasx2) [inline public/app/features/alerting/unified/components/amroutes/AmRootRouteForm.tsx:111]: Maybe put default as placeholder? See `description` above for what the default is

**Review by domasx2 — COMMENTED** [review]
looks good, few nits though :)

File: conf/provisioning/datasources/datasourceprovisioning.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+apiVersion: 1
```
Comment (peterholmberg) [inline conf/provisioning/datasources/datasourceprovisioning.yaml:1]: is this file added on purpose?

Comment (nathanrodman) [reply conf/provisioning/datasources/datasourceprovisioning.yaml]: shoot, nope. That was my Friday brain

**Review by peterholmberg — CHANGES_REQUESTED** [review]
Looks good! Just one issue with expand all.

File: public/app/features/alerting/unified/RuleList.tsx [hunk 3]
```diff
@@ -76,8 +77,19 @@ export const RuleList = withErrorBoundary(
...
+                    className={styles.expandAllButton}
+                    icon={expandAll ? 'angle-double-up' : 'angle-double-down'}
+                    variant="secondary"
+                    onClick={() => setExpandAll(!expandAll)}
+                  >
+                    {expandAll ? 'Collapse all' : 'Expand all'}
+                  </Button>
+                )}
+                <RuleStats showInactive={true} showRecording={true} namespaces={filteredNamespaces} />
+              </div>
```
[the remark this answers is not in the document]
Comment (nathanrodman) [reply public/app/features/alerting/unified/RuleList.tsx:92]: @peterholmberg that sounds like a good compromise. I also was running into performance issues when trying to expand and collapse with 1500 rules or so.

**Review by peterholmberg — APPROVED** [review]
Nice ✌️

**Pull request merged**